### PR TITLE
Aegisub: update to wangqr fork 3.3.2 (fix #31637)

### LINKED ISSUE #31637: Aegisub-3.2.2_12 doesn't load any audio
### System

```
$ xuname
Void 5.12.10_1 x86_64 GenuineIntel uptodate rrrrmdFFFFFF
$ xbps-query -p pkgver Aegisub
Aegisub-3.2.2_12
$ xbps-query -p pkgver libffms2
libffms2-2.40_1
```
### Expected behavior

Opening mp4, avi, mov, wav, mp3 or anything else that has audio should work.

### Actual behavior

FFmpeg says there are no audio frames:

image

> None of the available audio providers have a codec available to handle
> the selected file.
> 
> The following providers were tried:
> 
> FFmpegSource: Failed to open audio track: Audio track contains no
> audio frames

### Steps to reproduce the behavior

Run Aegisub and try to open a file that has audio in it (such as mp4, avi, mov, wav, mp3, ...)

### PR DESCRIPTION
**Pull request by pudiva**
~~This is a -git package because releases seem to be broken upstream and
don't build...~~

This package fetches from git because upstream doesn't support release
tarballs. The tarballs present on its github releases page are
auto-generated and fail to build with:

    checking for version... git repo not found and no cached git_version.h

See: https://github.com/wangqr/Aegisub/issues/116

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [x] I built this PR locally for my native architecture, (x86_64-glibc)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl

### CODE REVIEW

**Comment by abenson** [comment]
`-git` packages won't be expected.

**Comment by abenson** [comment]
Can the last release be built with specific patches?

**Comment by pudiva** [comment]
@Gottox might be interested since they're the maintainer of the main Aegisub pkg

**Comment by pudiva** [comment]
> `-git` packages won't be accepted.

Not even if it's super nice? Pretty please? :smiley: 

> Can the last release be built with specific patches?

I've failed so far. I could try tomorrow.

**Comment by pudiva** [comment]
Also @ericonr

**Comment by paper42** [comment]
Is there a reason why you chose to create a new package instead of updating the old one?

**Comment by pudiva** [comment]
> Is there a reason why you chose to create a new package instead of updating the old one?

Yes, because it's based on a fork. Should I update the original pkg instead?

**Comment by pudiva** [comment]
Okay I've managed to checkout and build tag v3.3.2 with CMake (which had failed previously with autotools). I've added a new commit to this PR, please tell me which one you prefer and I'll squash/drop properly.

**Comment by paper42** [comment]
> > Is there a reason why you chose to create a new package instead of updating the old one?
> 
> Yes, because it's based on a fork. Should I update the original pkg instead?

If the original upstream is dead, I believe this PR should just replace the original template.

File: srcpkgs/Aegisub-wangqr/template
Comment (paper42) [inline srcpkgs/Aegisub-wangqr/template:49]: It would be much better to download the release tarball and patch the version detection script. I don't think it will be accepted this way.

**Comment by ericonr** [comment]
> If the original upstream is dead, I believe this PR should just replace the original template.

Agreed, that was the intention of my original suggestion.

File: srcpkgs/Aegisub-wangqr/template
Comment (ericonr) [reply srcpkgs/Aegisub-wangqr/template:49]: We have packages that do this, for a myriad reasons, but yes, I'd rather not *move* to such a setup.

At worst, `echo $version > somefile` and patch cmake to read from that instead.

Comment (pudiva) [reply srcpkgs/Aegisub-wangqr/template:49]: > At worst, echo $version > somefile and patch cmake to read from that instead.

Unfortunately this is how **build/version.sh** generates a new **git_version.h** file:

```sh
new_version_h="\
#define BUILD_GIT_VERSION_NUMBER ${git_revision}
#define BUILD_GIT_VERSION_STRING \"${git_version_str}\"
#define TAGGED_RELEASE ${tagged_release}
#define INSTALLER_VERSION \"${installer_version}\"
#define RESOURCE_BASE_VERSION ${resource_version}"
```

I'm concerned with the `BUILD_GIT_VERSION_NUMBER` macro, that is set to `$git_revision`, that is defined as:

```sh
last_svn_revision=6962
last_svn_hash="16cd907fe7482cb54a7374cd28b8501f138116be"

git_revision=$(expr $last_svn_revision + $(git log --pretty=oneline $last_svn_hash..HEAD 2>/dev/null | wc -l))
```

which basically counts the number of commits since  `$last_svn_hash` and adds `$last_svn_revision`.

Now there are three options:

1. fetch with git
2. generate **git_version.h** on `pre_configure()`
   2.1. using hardcoded values manually extracted from git by the pkg maintainer (me???) on every update
   2.2. using the template's `$version` and never updating the other values

I did a little grepping but gave up following where the variable is ultimately used, so I'm not sure how important it is :woman_shrugging: 

To be honest, I'd rather go with what the maintainer supports, which is building from git, instead of creating more pkg maintenance overhead trying to build from the release tarballs that they'd have removed if they could:

> There are no release tarballs. “Source Code (zip)” and “Source Code (tar.gz)” listed on the release page are just automatically generated archives of the source, _without the `.git`-folder_ thus making them useless for regular building.Afaik GitHub doesn't allow to disable those.

https://github.com/wangqr/Aegisub/issues/116#issuecomment-886105613

... and I'm not even sure this is the only problem with the tarballs.

What do you think? :sparkles: :sparkles: :sparkles:

Comment (ericonr) [reply srcpkgs/Aegisub-wangqr/template:49]: Point against building from git is that we lose caching, so every build has to repeat the huge download. The reason we do it for `ardour` is because upstream deletes their pre-generated tarballs, but I still greatly dislike it...

IMO let's move the PR forward by pulling from git, then ponder more from there.

Please do like it's done in `ardour`, where we checkout a specific commit.

Comment (pudiva) [reply srcpkgs/Aegisub-wangqr/template:49]: > Please do like it's done in ardour, where we checkout a specific commit.

You mean you want me to reference the commit hash instead of tag v3.3.2?

Comment (pudiva) [reply srcpkgs/Aegisub-wangqr/template]: > > Please do like it's done in ardour, where we checkout a specific commit.
> 
> You mean you want me to reference the commit hash instead of tag v3.3.2?

I did that already. Please, take a look and tell me what you think. :sparkles:

File: srcpkgs/Aegisub/template
Comment (paper42) [inline srcpkgs/Aegisub/template:6]: this wrksrc is default, so no need to overwrite it

File: srcpkgs/Aegisub/template
Comment (paper42) [inline srcpkgs/Aegisub/template:32]: we usually have multiple dependencies on one line and wrap at 80 columns

File: srcpkgs/Aegisub/template
Comment (paper42) [inline srcpkgs/Aegisub/template:33]: is the wanqr fork very different from the original or is it just a maintained continuation of the original? In that case, I don't think we would need to include (wangqr fork)

File: srcpkgs/Aegisub/template
Comment (paper42) [inline srcpkgs/Aegisub/template:37]: this is an old artifact from when this was a separate package, now Aegisub shouldn't conflict with Aegisub

File: srcpkgs/Aegisub/template
Comment (paper42) [inline srcpkgs/Aegisub/template:57]: the added line is commented out, so it doesn't do anything

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template:33]: I'm not sure. I'd keep it anyway so it's easy to see which fork we're using, as there might be others around.

Comment (ericonr) [reply srcpkgs/Aegisub/template:20]: That information can usually be had from other sources, like the homepage :)

I'd prefer to leave the description lean.

File: srcpkgs/Aegisub/template
Comment (ericonr) [inline srcpkgs/Aegisub/template:19]: Multiline strings should be indented with spaces, and don't leave the terminating quote in a new line.

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template]: Okayy, done!

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template]: done!

Comment (pudiva) [reply srcpkgs/Aegisub/template]: okay looks like the linter wants me to indent with one space, not two...

**Comment by pudiva** [comment]
1 week later... looks like GitHub Actions is offline atm. Will look again tomorrow if CI is passing. :)

File: srcpkgs/Aegisub/template
Comment (ericonr) [inline srcpkgs/Aegisub/template:43]: Do you know if this is still true?

File: srcpkgs/Aegisub/template
Comment (ericonr) [inline srcpkgs/Aegisub/template:30]:
```suggestion
	configure_args+=" -DwxWidgets_CONFIG_OPTIONS='--prefix=${XBPS_CROSS_BASE}'"
	hostmakedepends+=" wxWidgets-gtk3-devel"
```

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template:43]: I didn't test on Wayland, but it seems it's not! Thanks! :sparkles:

https://github.com/wangqr/Aegisub/blob/723d01d13088d9b1bccc821f1129d045a0b74e2c/packages/desktop/aegisub.desktop.template.in#L7

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template:30]: ooops :upside_down_face:

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template]: done :sparkles:

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template]: fixed :sparkles:

**Comment by pudiva** [comment]
> For fixing cross, you might want to take a look at how kicad does it, maybe. It uses our CMake wx helper.

@ericonr looks like cross is still broken and I don't know how to fix it :(

**Comment by pudiva** [comment]
```
[8/328] Linking C executable luajit-minilua
[9/328] Generating vendor/luajit/src/gen/buildvm_arch.h
FAILED: vendor/luajit/src/gen/buildvm_arch.h /builddir/Aegisub-3.3.2/BUILD/vendor/luajit/src/gen/buildvm_arch.h 
cd /builddir/Aegisub-3.3.2/vendor/luajit/src && /usr/bin/cmake -E make_directory /builddir/Aegisub-3.3.2/BUILD/vendor/luajit/src/gen && luajit-minilua ../dynasm/dynasm.lua -D P64 -D JIT -D FFI -D FPU -D HFABI -D VER= -o /builddir/Aegisub-3.3.2/BUILD/vendor/luajit/src/gen/buildvm_arch.h vm_x86.dasc
/bin/sh: line 1: luajit-minilua: command not found
```

so looks like it builds `luajit-minilua` and then tries to use it, which probably fails because host can't run target binaries?

I know there's an option to use system lua, but I had problems with it before. I think it has to do with the version void ships.

Mind explaining why cross compiling is important here, @ericonr ? :slightly_smiling_face:

**Comment by pudiva** [comment]
> > so looks like it builds luajit-minilua and then tries to use it, which probably fails because host can't run target binaries?
> 
> More specifically, it fails to find the binary's dynamic linker, which makes for quite a terrible error message. But yes :)

Sorry, this is indeed a terribly misleading error message that I have gotten used to and sometimes forget how not obvious it can be to others ~ thanks for explaining :sparkles: 

> > I know there's an option to use system lua, but I had problems with it before. I think it has to do with the version void ships.
> 
> Do you remember the problem you had with it? Void ships a bunch of Lua versions...

The old template had the option to use system Lua, but it failed because some Aegisub script uses a syntax that is incompatible with the version void ships... or at least that's what I remember.

Anyway, that was with the problematic `./configure` script, the CMake doesn't seem to have the option of using system Lua at all.

> > Mind explaining why cross compiling is important here, @ericonr ? slightly_smiling_face
> 
> Mostly so we can build it for ARM/aarch64 :) technically it's broken for cross already, so adding `nocross` would be ok, but it's always nice to improve a situation :p

Cool, so I'll try a bit more later today and then add the `nocross` if everything fails. :)

**Comment by pudiva** [comment]
Ok so I couldn't fix cross, but I could remove some deps that weren't necessary anymore: LuaJIT, automake and libtool.

**Comment by pudiva** [comment]
bump? :upside_down_face:

**Review by paper42 — COMMENTED** [review]
the commit title should be something like `Aegisub: update to 3.3.2, switch to a maintained fork`. If it's too long, mentioning moving distfiles to the fork could be in the commit body.

It still probably doesn't work on Wayland, so I think we should still force X11
```
Fatal Error: wxGLCanvas is only supported on X11 currently.  You may be able to
work around this by setting environment variable GDK_BACKEND=x11 before starting
your program.
```

**Review by paper42 — APPROVED** [review]
lgtm, I will leave this to @ericonr

**Comment by pudiva** [comment]
@ericonr bump :upside_down_face:

File: srcpkgs/Aegisub/template
Comment (ericonr) [inline srcpkgs/Aegisub/template:39]: I think you forgot to remove this code block?

**Comment by ericonr** [comment]
And please import https://github.com/wangqr/Aegisub/pull/121 ! It should allow us to build from tarball instead of git clone :)

**Comment by ericonr** [comment]
Ah, btw, to use that, you do `export FORCE_GIT_VERSION=$version` in `pre_build`

File: srcpkgs/Aegisub/template
Comment (pudiva) [reply srcpkgs/Aegisub/template:39]: I re-added it as @paper42 said it was still broken on wayland https://github.com/void-linux/void-packages/pull/32162#pullrequestreview-737696497

**Comment by pudiva** [comment]
> And please import [wangqr/Aegisub#121](https://github.com/wangqr/Aegisub/pull/121) ! It should allow us to build from tarball instead of git clone :)

do you mean cherry-pick this patch and apply to the latest tarball?

btw thanks for the patch! :sparkles::sparkles::sparkles:

File: srcpkgs/Aegisub/template
Comment (ericonr) [reply srcpkgs/Aegisub/template:39]: Sure, but the file in https://github.com/wangqr/Aegisub/blob/dev/packages/desktop/aegisub.desktop.template.in already  applies the variable (if you switch to using `vsed` there it will print a warning that it didn't replace anything).

Comment (pudiva) [reply srcpkgs/Aegisub/template:39]: ohhh sorry my confusion

Comment (ericonr) [reply srcpkgs/Aegisub/template:39]: no worries ;)

**Comment by pudiva** [comment]
@ericonr I think it's good now.

I had to resolve a merge conflict with your patch as it didn't apply to the tarball ~ I think it's fine, but a quick look wouldn't hurt. :eyes:

Cheers! :slightly_smiling_face:

**Comment by ericonr** [comment]
Updated commit message, merging.

**Comment by q66** [comment]
i don't approve of the way this PR just silently drops support for building with system luajit

it was doing that for a reason, and now a whole bunch of platforms are just broken (same with the libatomic bits)

**Pull request merged**
